deeppavlov/core/layers/embeddings.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2017 Neural Networks and Deep Learning lab, MIPT
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Union, Sequence
+
+import tensorflow as tf
+from tensorflow.python.eager import context
+from tensorflow.python.keras.utils import tf_utils
+from tensorflow.python.framework import ops, tensor_shape
+from tensorflow.python.ops import embedding_ops
+
+
+class AdvancedEmbedding(tf.keras.layers.Layer):
```

Comment:
Bad naming. It is by far not clean from it what is the purpose of the class.

Reply:
Maybe BERTEmbedder is a better name

Reply:
And should it satisfy interface for deeppavlov embedder?

Reply:
BERTEmbedder was the name of this class some commits ago... I am not sure whether it is only BERT-specific embeddings: without segment embeddings, it looks like common for transformers. The current name is definitely better than EmbeddingPostprocessor (https://github.com/tensorflow/models/blob/master/official/bert/modeling.py#L254).
Breaking down this class into three distinct could simplify naming but I doubt their use cases will go further than usage in the BERT.
Let's wait for some thoughts from other reviewers.

Reply:
If you doubt that use cases will go further than usage in the BERT, just name it BERT embedder.